### ISSUE TITLE
Get the basedir reliably

### ISSUE THREAD

**Issue by ppalaga** [apache/camel-quarkus#439]
A followup of https://github.com/apache/camel-quarkus/pull/440 : 
In XsltProcessor we guess the basedir via ArchiveRootBuildItem..getArchiveRoot().getParent().getParent() which works for Maven, but would probably fail for Gradle and other build systems. Let's see if we can get this from Quarkus. Blocked by https://github.com/quarkusio/quarkus/issues/5578

**Comment by ppalaga** [comment]
This Quarkus PR https://github.com/quarkusio/quarkus/pull/5478 adds the necessary `ModuleDirBuildItem`

**Comment by lburgazzoli** [comment]
This relates to https://github.com/apache/camel-quarkus/pull/451 so not something that is in the code-base. We should avoid opening issues on code not in master as it is misleading.

**Comment by ppalaga** [comment]
> This relates to #451 so not something that is in the code-base. We should avoid opening issues on code not in master as it is misleading.

I admit I created this issue before the underlying code was merged although it does not relate to #451. It relates to #440 (as stated in the first post) that was merged 2 days ago. I did that because I wanted to paste the URL of this issue directly to #440 https://github.com/apache/camel-quarkus/pull/440/files#diff-acae606a994acbe2ecb513085c1b01e5R93 But OK, can avoid creating followup issues prematurely.

**Comment by ppalaga** [comment]
So we won't get a baseDir BuildItem from quarkus because

> one of our [Quarkus] use cases is to be able augment existing jars (e.g. to provide security updates by building an app against an updated version of the Quarkus platform). This build item would break that use case if build steps start relying on it.

As consequence of that we agreed with @lburgazzoli to stop supporting all but classpath XSLT URIs. Let me file a separate issue for that.
